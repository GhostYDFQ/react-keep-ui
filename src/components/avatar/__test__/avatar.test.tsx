import React from "react";
import { render, cleanup } from "@testing-library/react";
import { Avatar, AvatarSize } from "../index";

describe("test Avatar component", () => {
    it("it should render default avatar", () => {
        let wrapper = render(<Avatar data-testid="avatar-div"></Avatar>);
        expect(wrapper).toMatchSnapshot();
        let div = wrapper.getByTestId("avatar-div");
        expect(div).toBeInTheDocument();
        let username = wrapper.getByText("l");
        expect(username).toBeTruthy();
    });

    it("should correct loading", () => {
        let wrapper = render(<Avatar data-testid="avatar-div" isLoading></Avatar>);
        expect(wrapper).toMatchSnapshot();
        let div = wrapper.getByTestId("avatar-div");
        let svg = div.firstChild;
        expect(svg).toBeVisible();
        cleanup();

        wrapper = render(
            <Avatar data-testid="avatar-div" isLoading username="123" src="/" size="tiny"></Avatar>
        );
        div = wrapper.getByTestId("avatar-div");
        svg = div.firstChild;
        expect(svg).toBeVisible();
    });

    it("should correct img", () => {
        let wrapper = render(<Avatar data-testid="avatar-div" src="www.test.com"></Avatar>);
        let div = wrapper.getByTestId("avatar-div");
        let img = div.firstChild;
        expect(img.tagName).toEqual("IMG");
        expect(img).toHaveAttribute("src", "www.test.com");
        expect(img).toHaveAttribute("alt", "loading");
        cleanup();

        wrapper = render(
            <Avatar data-testid="avatar-div" src="www.yezun.xyz" username="yezun"></Avatar>
        );
        div = wrapper.getByTestId("avatar-div");
        img = div.firstChild;
        expect(img).toHaveAttribute("src", "www.yezun.xyz");
        expect(img).toHaveAttribute("alt", "yezun");
    });

    it("should render correct username", () => {
        let wrapper = render(<Avatar data-testid="avatar-div" username="yezun"></Avatar>);
        expect(wrapper).toMatchSnapshot();
        let div = wrapper.getByTestId("avatar-div");
        let username = wrapper.getByText("y");
        expect(username).toBeVisible();
        cleanup();

        wrapper = render(<Avatar username="中文汉字"></Avatar>);
        username = wrapper.getByText("中");
        expect(username).toBeTruthy();
    });
});
