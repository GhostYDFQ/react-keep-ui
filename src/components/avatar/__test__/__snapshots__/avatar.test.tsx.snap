// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Avatar component it should render default avatar 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-dlnjPT gpBaeP"
        data-testid="avatar-div"
      >
        <div
          aria-hidden="true"
          class="sc-hKFyIo dXXoME"
        >
          l
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-dlnjPT gpBaeP"
      data-testid="avatar-div"
    >
      <div
        aria-hidden="true"
        class="sc-hKFyIo dXXoME"
      >
        l
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`test Avatar component should correct loading 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-dlnjPT lhACca"
        data-testid="avatar-div"
      >
        <svg
          class="sc-bdnylx fIWMCQ"
          color="black"
          height="20px"
          viewBox="0 0 1024 1024"
          width="20px"
        >
          <path
            class="sc-gtssRu kxOhqr"
            color="black"
            d="M533 960a850 850 0 0 0 386-92v-19c0-117-242-223-306-234-20-3-21-58-21-58s59-58 72-137c35 0 56-84 21-113 2-31 45-243-173-243S337 276 338 307c-34 29-13 113 22 113 13 79 72 137 72 137s-1 55-21 58c-64 11-301 115-306 231a855 855 0 0 0 428 114z"
          />
        </svg>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-dlnjPT lhACca"
      data-testid="avatar-div"
    >
      <svg
        class="sc-bdnylx fIWMCQ"
        color="black"
        height="20px"
        viewBox="0 0 1024 1024"
        width="20px"
      >
        <path
          class="sc-gtssRu kxOhqr"
          color="black"
          d="M533 960a850 850 0 0 0 386-92v-19c0-117-242-223-306-234-20-3-21-58-21-58s59-58 72-137c35 0 56-84 21-113 2-31 45-243-173-243S337 276 338 307c-34 29-13 113 22 113 13 79 72 137 72 137s-1 55-21 58c-64 11-301 115-306 231a855 855 0 0 0 428 114z"
        />
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`test Avatar component should render correct username 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-dlnjPT gpBaeP"
        data-testid="avatar-div"
      >
        <div
          aria-hidden="true"
          class="sc-hKFyIo dXXoME"
        >
          y
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-dlnjPT gpBaeP"
      data-testid="avatar-div"
    >
      <div
        aria-hidden="true"
        class="sc-hKFyIo dXXoME"
      >
        y
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
